stl/inc/valarray

```diff
@@ -1372,6 +1372,11 @@ public:
     _NODISCARD size_t stride() const noexcept /* strengthened */ {
         return _Stride;
     }
+#if _HAS_CXX200
+    friend bool operator==(const slice& x, const slice& y) {

```

Comment:
In the STL's implementation, we're required to write code differently in several ways from what's depicted in the Standard (and what all other programmers write). The biggest way is that we can't use ordinary identifiers like `x` and `y` for parameters, local variables, etc. because the STL is required to be extremely resistant to user-defined macros and other things. So, we need to use identifiers that the Standard reserves exclusively for use by the compiler and Standard Library. We conventionally use `_Leading_underscore_capital` identifiers, which we refer to as `_Ugly`. (`__double_underscore_anywhere` identifiers are also reserved, but we use them in only a few places.) Additionally, we avoid identifiers of the form underscore-capital because although `_X` and `_Y` are reserved, they look too much like macros (and `_T` is actually macroized by Windows headers).

Common names for comparison parameters are `_Left` and `_Right` or `_Lhs` and `_Rhs`.